jwst/cube_build/blot_cube_build.py

```diff
@@ -281,19 +231,135 @@ def blot_overlap_quick(self, model, xcenter, ycenter, xstart,
             if len(index_x[0]) > 0 and len(index_y[0]) > 0:
                 d1pix = np.array(x_cube[ipt] - xcenter[index_x])
                 d2pix = np.array(y_cube[ipt] - ycenter[index_y])
-                dxy=[]
-                for dy in d2pix:
-                    for dx in d1pix:
-                        dxy.append((dx * dx) + (dy * dy))
+
+                dxy = [ (dx*dx + dy*dy)  for dy in d2pix for dx in d1pix]
                 dxy = np.sqrt(dxy)
                 weight_distance = np.exp(-dxy)
                 weighted_flux = weight_distance * flux_cube[ipt]
                 index2d = [iy * blot_xsize + ix for iy in index_y[0] for ix in (index_x[0]+xstart)]
                 blot_flux[index2d] = blot_flux[index2d] + weighted_flux
                 blot_weight[index2d] = blot_weight[index2d] + weight_distance
 
+        t1 = time.time()
+        log.debug("Time to blot median image to input model = %.1f s" % (t1-t0,))
         # done mapping blotted x,y (x_cube, y_cube) to detector
         igood = np.where(blot_weight > 0)
         blot_flux[igood] = blot_flux[igood] / blot_weight[igood]
         blot_flux = blot_flux.reshape((blot_ysize, blot_xsize))
         return blot_flux
+
+    # ************************************************************************
+    def blot_images_nirspec(self):
+        """ Core blotting routine for NIRSPEC
+
+        This is the main routine for blotting the NIRSPEC median sky image back to
+        the detector space and creating a blotting image for each input model.
+        This routine was split from the MIRI routine because the blotting for NIRSpec
+        needs to be done slice by slice and an error in the inverse mapping (sky to
+        detector) mapped too many values back to the detector. This routine adds
+        a check and first pulls out the min and max ra and dec values in the slice
+        and only inverts the slice values back to the detector.
+        1. Loop over every data model to be blotted and find ra,dec,wavelength
+           for every pixel in a valid slice on the detector.
+        2. For each slice get the WCS info for the slice. Find the x,y bounding
```

Comment:
It sounds like ra, dec, wavelength values were found in step 1. Are they being found/computed again in step 2? Or just using the values from step 1 to find the min/max?

Reply:
rewrote section